File: frontend/src/app/features/dashboard/transaction/transaction-dto.ts
import {TransactionStatus} from '../../../shared/transaction-status';

export interface TransactionDto {
  id: number;
  sellerId: number;
  sellerName: string | null;
  buyerId: number;
  buyerName: string | null;
  invoiceNumber: string | null;
  amount: number;
  discount: number;
  description: string | null;
  fulfillmentDate: string | null;
  sellerApproved: string | null;
  sellerApprover: number | null;
  sellerApproverName: string | null;
  buyerApproved: string | null;
  buyerApprover: number | null;
  buyerApproverName: string | null;
  status: TransactionStatus;
  createdAt: string;
  modifiedAt: string;
}

File: backend/src/main/java/hu/ibc/ibcpartners/partner/dto/TransactionDto.java
package hu.ibc.ibcpartners.partner.dto;

import hu.ibc.ibcpartners.partner.entity.TransactionStatus;
import jakarta.validation.constraints.NotNull;

import java.time.Instant;
import java.time.LocalDate;

public record TransactionDto(
        Long id,
        @NotNull Long sellerId,
        String sellerName,
        @NotNull Long buyerId,
        String buyerName,
        String invoiceNumber,
        @NotNull Long amount,
        @NotNull Long discount,
        String description,
        LocalDate fulfillmentDate,
        Instant sellerApproved,
        Integer sellerApprover,
        String sellerApproverName,
        Instant buyerApproved,
        Integer buyerApprover,
        String buyerApproverName,
        TransactionStatus status,
        Instant createdAt,
        Instant modifiedAt
) {
}
